tensorboard/plugins/base_plugin.py

```diff
@@ -97,6 +97,16 @@ def frontend_metadata(self):
         """
         return FrontendMetadata()
 
+    def relevant_summary_plugins(self):
```

Comment:
Naming nit: maybe something more like `data_plugin_names()`?

"Relevant" seems a little unnecessary IMO in that this is already clearly plugin-specific and the fact that it's plural already implies that it's not necessarily just this plugin's own name.

I'd maybe avoid "summary" because in the long run we don't necessarily consider everything exposed via the data provider to be "summary" data per se.  E.g. profile data might get exposed via the data provider API at some point, or projector data, and while those are maybe sort of still conceptually connected to the `tf.summary` API, they're never materialized as a literal `tf.Summary` proto so it seems a little odd to label them as such.  (I also just don't love the term "summary" in general since it's so disconnected from the normal English meaning of the word, so to the extent we can get away with other terminology I tend to favor that.)